src/data/nat/enat.lean

```diff
@@ -361,6 +361,14 @@ begin
   apply_mod_cast nat.lt_succ_of_le, apply_mod_cast h
 end
 
+lemma ne_zero_iff_one_le {n : enat} : n ≠ 0 ↔ 1 ≤ n :=
```

Comment:
I think we're missing a bit of API here. We should first create `ne_bot_iff : x ≠ ⊥ ↔ ⊥ < x`. Then, we could create the lemmas `eq_zero_iff` and `ne_zero_iff` on `enat` to be def-eq to `eq_bot_iff` and `ne_bot_iff`.

Then, we should define a `succ_order` instance on `enat`. This could be done immediately via `order.with_top.succ_order_of_no_max_order` if only `enat` were defined as `with_top ℕ`. Apparently this will be a complicated refactor, see [Zulip](https://leanprover.zulipchat.com/#narrow/stream/113488-general/topic/enat.20pain). 

We'll also need a lemma `not_is_max_coe : ¬ is_max (↑ n)` on `enat`.

With all this done, and with the help of a few lemmas I'm adding on #15016, we could prove `succ_zero : succ 0 = 1` (it might even be def-eq). Finally, the proof of this theorem would just be 
```lean
rw [ne_bot_iff, ←succ_zero, succ_le_iff_of_not_is_max, lt_succ_iff_of_not_is_max (not_is_max_coe 0)]
```
Obviously it's completely unreasonable to hold this PR up because of this. I just mention it to highlight what we're still missing.

Reply:
#15029 for a few of those missing lemmas I mention.